dbcon/ddlpackage/ddl.y

```diff
@@ -56,24 +56,42 @@ int ddllex(YYSTYPE* ddllval, void* yyscanner);
 void ddlerror(struct pass_to_bison* x, char const *s);
 char* copy_string(const char *str);
 
-void fix_column_length(SchemaObject* elem, const CHARSET_INFO* def_cs) {
+void fix_column_length(SchemaObject* elem, const CHARSET_INFO* def_cs)
+{
     auto* column = dynamic_cast<ColumnDef*>(elem);
+
     if (column == NULL || column->fType == NULL)
     {
         return;
     }
 
+    if (column->fType->fType == DDL_BLOB ||
```

Comment:
The name of the function is misguiding. fix_column_length() starts to assign charset related stuff out of a sudden.

Reply:
I suggest to rename the function or put the functionality into a separate function.

Reply:
Good point. I have renamed the function.